src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -933,6 +983,31 @@ private static RepairSegment createRepairSegmentFromRow(Row segmentRow) {
     return Optional.empty();
   }
 
+  @Override
+  public Optional<RepairSegment> getNextFreeSegmentForRanges(
+      UUID runId, Optional<RingRange> parallelRange, List<RingRange> ranges) {
+    List<RepairSegment> segments
+        = Lists.<RepairSegment>newArrayList(getRepairSegmentsForRun(runId));
+    Collections.shuffle(segments);
+
+    for (RepairSegment seg : segments) {
+      if (seg.getState().equals(State.NOT_STARTED) && withinRange(seg, parallelRange)) {
```

Comment:
how do we handle  `seg.getState().equals(State.NOT_STARTED)`  changing to STARTED by someone else?

Reply:
In this case the segment will have a leader already and the repair manager will move to the next one.

Reply:
Wouldn't it be cheaper to first check against `STARTED` ? (it is meant as an optimisation)

Reply:
With the lock that is applied in the sidecar mode, this kind of conflict couldn't even happen actually. For the other modes, the optimization would be very marginal I think, not worth the time to code and test.